src/util.rs

```diff
@@ -0,0 +1,43 @@
+//! Common logic for interacting with remote object stores
+use super::Result;
+use bytes::Bytes;
+use futures::{stream::StreamExt, Stream};
+
+/// Returns the prefix to be passed to an object store
+#[cfg(any(feature = "aws", feature = "gcp", feature = "azure"))]
+pub fn format_prefix(prefix: Option<&crate::path::Path>) -> Option<String> {
+    prefix
+        .filter(|x| !x.as_ref().is_empty())
+        .map(|p| format!("{}{}", p.as_ref(), crate::path::DELIMITER))
+}
+
+/// Returns the range to be passed to an object store
+#[cfg(any(feature = "aws"))]
+pub fn format_range(range: std::ops::Range<usize>) -> String {
+    format!("bytes={}-{}", range.start, range.end.saturating_sub(1))
+}
```

Comment:
This is AWS-specific, no? If it's specific, it should probably live within the AWS module. If it's not AWS-specific but follows some common HTTP standard, it would be nice to maybe rename the function to `format_http_range` and cite a standard.